src/components/Dropdown/Dropdown.tsx

```diff
@@ -0,0 +1,111 @@
+import React, {
+  Dispatch,
+  ReactElement,
+  ReactNode,
+  SetStateAction,
+} from "react";
+import styled, { css, useTheme } from "styled-components";
+import { Menu } from "@headlessui/react";
+import { IconDirection } from "../Icon/Direction";
+import { Text } from "../../typography";
+import { Button } from "../Button";
+import Icon from "../Icon";
+
+export interface DropdownOption {
+  value: string;
+  label: string;
+}
+
+export interface DropdownProps {
+  buttonContent: ReactNode;
+  options: DropdownOption[];
+  selected?: DropdownOption;
+  setSelected: Dispatch<SetStateAction<DropdownOption | undefined>>;
+}
+
+export const Dropdown = ({
+  buttonContent,
+  options,
+  selected,
+  setSelected,
+}: DropdownProps): ReactElement | null => {
+  const theme = useTheme();
+
+  const isDropdownOptionSelected = ({ value }: DropdownOption): boolean => {
+    return value === selected?.value;
+  };
+
+  return (
+    <Menu>
+      {({ open }) => (
+        <>
+          <StyledMenuButton>
+            {buttonContent}
+            <ChevronComponent open={open} />
+          </StyledMenuButton>
+          <StyledMenuItems>
+            {options.map((option) => (
+              <Menu.Item key={option.value} onClick={() => setSelected(option)}>
+                {({ active }: { active: boolean }) => (
+                  <StyledMenuItem
+                    active={active || isDropdownOptionSelected(option)}
+                    color={theme.colors.primary}
```

Comment:
since this is directly related to theme, could we use the `args` from styled component directly to not have to deal with css on the react component?

see: https://styled-components.com/docs/basics#attaching-additional-props

Reply:
Good catch, I think initially I had a few other props there but then refactored and was left only with color.

We actually don't even need to attach additional props because we have access to the theme prop, so it is enough to just do this https://github.com/Major-Eth/yearn-web-lib/pull/28/commits/90de9f0e25b9f0233370420e137f5df347e1e83a